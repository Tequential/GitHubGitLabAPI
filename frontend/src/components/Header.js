//Header to display loading and/or errors
export function Header({ loading, error }) {
    return (
      <div className="App-header" >
          Github/GitLab Users
          {loading && <div>A moment please...</div>}
          {error && (
            <div className="error">{`There is a problem fetching data from the API - ${error}`}</div>
          )}
      </div>
    );
  }

  export default Header;