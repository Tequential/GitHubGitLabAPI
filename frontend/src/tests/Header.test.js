import { render } from '@testing-library/react';
import Header from '../components/Header';
import renderer from 'react-test-renderer';

test('renders header correctly', () => {
  render(<Header />);
  const tree = renderer.create(<Header/>).toJSON();
  expect(tree).toMatchSnapshot();
});