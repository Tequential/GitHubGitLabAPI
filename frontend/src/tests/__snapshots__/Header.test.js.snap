// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header correctly 1`] = `
<div
  className="App-header"
>
  Github/GitLab Users
</div>
`;
